src/docfx/restore/Restore.cs

```diff
@@ -17,22 +17,24 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Repo
         {
             // Restore has to use Config directly, it cannot depend on Docset,
             // because Docset assumes the repo to physically exist on disk.
+            var errors = new List<Error>();
```

Comment:
this is not thread safe if `RestoreDocset` runs in parallel.

Reply:
Right, will update it to ConcurrentBag

---
In reply to: [211463029](https://github.com/dotnet/docfx/pull/3243#discussion_r211463029) [](ancestors = 211463029)

Reply:
Pass Report in RestoreDocset instead

---
In reply to: [211463509](https://github.com/dotnet/docfx/pull/3243#discussion_r211463509) [](ancestors = 211463509,211463029)